Compilation error if compiling against system openal-soft library

When building the game with `USE_SYSTEM_LIBS=ON`, I ran into following errors:

```
[ 78%] Built target fgame
[ 78%] Building CXX object CMakeFiles/openmohaa.dir/code/uilib/uiwinman.cpp.o
[ 78%] Building CXX object CMakeFiles/openmohaa.dir/code/client/snd_openal_new.cpp.o
[ 79%] Building C object CMakeFiles/openmohaa.dir/code/client/snd_codec_mp3.c.o
[ 79%] Building CXX object CMakeFiles/openmohaa.dir/code/uilib/uistatus.cpp.o
[ 80%] Building CXX object CMakeFiles/openmohaa.dir/code/uilib/uivertscroll.cpp.o
[ 80%] Building CXX object CMakeFiles/openmohaa.dir/code/uilib/uiwidget.cpp.o
/home/tle/Work/openmohaa/code/client/snd_openal_new.cpp: In function ‘bool S_OPENAL_InitContext()’:
/home/tle/Work/openmohaa/code/client/snd_openal_new.cpp:444:39: error: ‘qalGetStringiSOFT’ was not declared in this scope; did you mean ‘qalGetString’?
  444 |         const ALchar* resamplerName = qalGetStringiSOFT(AL_RESAMPLER_NAME_SOFT, i);
      |                                       ^~~~~~~~~~~~~~~~~
      |                                       qalGetString
make[2]: *** [CMakeFiles/openmohaa.dir/build.make:1479: CMakeFiles/openmohaa.dir/code/client/snd_openal_new.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:440: CMakeFiles/openmohaa.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
Exit 2
```

Upon closer look into `code/client/qal.h`, I could see:

```c
#ifdef USE_OPENAL_DLOPEN
...
#ifdef AL_SOFT_source_resampler
extern LPALGETSTRINGISOFT qalGetStringiSOFT;
#endif
...

#endif
```

that means `qalGetStringiSOFT` definition only happen if `USE_OPENAL_DLOPEN` is defined. However it is not the case, according to `code/client/CMakeLists.txt`:

```cmake
if(NOT USE_SYSTEM_LIBS)
  target_compile_definitions(omohclient INTERFACE USE_OPENAL_DLOPEN=1)
else()
  # Link against system OpenAL
  target_link_libraries(omohclient INTERFACE OpenAL::OpenAL)
endif()
```

the `USE_OPENAL_DLOPEN` only get defined when `USE_SYSTEM_LIBS=OFF`

Therefore I believe this issue is NOT explicitly about a particular old version of openal-soft, in fact it could happen with any version.

--- a/code/client/qal.c
+++ b/code/client/qal.c
@@ -88,9 +88,6 @@ LPALGETBUFFERI qalGetBufferi;
 LPALDOPPLERFACTOR qalDopplerFactor;
 LPALSPEEDOFSOUND qalSpeedOfSound;
 LPALDISTANCEMODEL qalDistanceModel;
-#ifdef AL_SOFT_source_resampler
-LPALGETSTRINGISOFT qalGetStringiSOFT;
-#endif
 
 LPALCCREATECONTEXT qalcCreateContext;
 LPALCMAKECONTEXTCURRENT qalcMakeContextCurrent;
@@ -212,9 +209,6 @@ qboolean QAL_Init(const char *libname)
 	qalDopplerFactor = GPA("alDopplerFactor");
 	qalSpeedOfSound = GPA("alSpeedOfSound");
 	qalDistanceModel = GPA("alDistanceModel");
-#ifdef AL_SOFT_source_resampler
-	qalGetStringiSOFT = GPA("alGetStringiSOFT");
-#endif
 
 	qalcCreateContext = GPA("alcCreateContext");
 	qalcMakeContextCurrent = GPA("alcMakeContextCurrent");

--- a/code/client/qal.h
+++ b/code/client/qal.h
@@ -135,9 +135,6 @@ extern LPALGETBUFFERIV qalGetBufferiv;
 extern LPALDOPPLERFACTOR qalDopplerFactor;
 extern LPALSPEEDOFSOUND qalSpeedOfSound;
 extern LPALDISTANCEMODEL qalDistanceModel;
-#ifdef AL_SOFT_source_resampler
-extern LPALGETSTRINGISOFT qalGetStringiSOFT;
-#endif
 
 extern LPALCCREATECONTEXT qalcCreateContext;
 extern LPALCMAKECONTEXTCURRENT qalcMakeContextCurrent;

--- a/code/client/snd_openal_new.cpp
+++ b/code/client/snd_openal_new.cpp
@@ -128,6 +128,13 @@ static ALuint S_OPENAL_Format(float width, int channels);
 #    define ALDRIVER_DEFAULT "libopenal.so.1"
 #endif
 
+//
+// alext
+//
+#ifdef AL_SOFT_source_resampler
+LPALGETSTRINGISOFT qalGetStringiSOFT;
+#endif
+
 /*
 ==============
 __alDieIfError
@@ -268,8 +275,6 @@ static bool S_OPENAL_InitContext()
 {
     const char *dev;
     int         attrlist[10];
-    size_t      i;
-    size_t      numResamplers;
 
     Com_DPrintf("OpenAL: Context initialization\n");
 
@@ -427,23 +432,6 @@ static bool S_OPENAL_InitContext()
     qalcMakeContextCurrent(al_context_id);
     alDieIfError();
 
-#ifdef AL_SOFT_source_resampler
-    al_default_resampler_index = qalGetInteger(AL_DEFAULT_RESAMPLER_SOFT);
-    alDieIfError();
-    al_resampler_index = al_default_resampler_index;
-    numResamplers = qalGetInteger(AL_NUM_RESAMPLERS_SOFT);
-    alDieIfError();
-
-    for (i = 0; i < numResamplers; i++) {
-        const ALchar* resamplerName = qalGetStringiSOFT(AL_RESAMPLER_NAME_SOFT, i);
-        if (Q_stristr(resamplerName, "spline")) {
-            Com_Printf("OpenAL: Using %s as the resampler.\n", resamplerName);
-            al_resampler_index = i;
-            break;
-        }
-     }
-#endif
-
     Com_Printf("AL_VENDOR: %s\n", qalGetString(AL_VENDOR));
     alDieIfError();
 
@@ -472,11 +460,7 @@ S_OPENAL_InitExtensions
 */
 static bool S_OPENAL_InitExtensions()
 {
-    ima4_ext         = qalIsExtensionPresent("AL_EXT_IMA4");
-    soft_block_align = qalIsExtensionPresent("AL_SOFT_block_alignment");
-
-    return true;
-
+    /*
     extensions_table_t extensions_table[4] = {
         "alutLoadMP3_LOKI",
         (void **)&_alutLoadMP3_LOKI,
@@ -488,6 +472,17 @@ static bool S_OPENAL_InitExtensions()
         (void **)&_alReverbDelay_LOKI,
         true
     };
+    */
+
+    extensions_table_t extensions_table[] = {
+#ifdef AL_SOFT_source_resampler
+        extensions_table_t {
+                            "alGetStringiSOFT", (void **)&qalGetStringiSOFT,
+                            false, },
+#endif
+        extensions_table_t {NULL, NULL, NULL}
+    };
+
     extensions_table_t *i;
 
     for (i = extensions_table; i->funcname; ++i) {
@@ -509,6 +504,9 @@ static bool S_OPENAL_InitExtensions()
 
         Com_Printf("...found.\n");
     }
+
+    ima4_ext         = qalIsExtensionPresent("AL_EXT_IMA4");
+    soft_block_align = qalIsExtensionPresent("AL_SOFT_block_alignment");
 
     qalGetError();
     return true;
@@ -629,6 +627,28 @@ qboolean S_OPENAL_Init()
     al_current_volume = Square(s_volume->value);
     qalListenerf(AL_GAIN, al_current_volume);
     alDieIfError();
+
+#ifdef AL_SOFT_source_resampler
+    if (qalGetStringiSOFT) {
+        size_t numResamplers;
+        size_t i;
+
+        al_default_resampler_index = qalGetInteger(AL_DEFAULT_RESAMPLER_SOFT);
+        alDieIfError();
+        al_resampler_index = al_default_resampler_index;
+        numResamplers      = qalGetInteger(AL_NUM_RESAMPLERS_SOFT);
+        alDieIfError();
+
+        for (i = 0; i < numResamplers; i++) {
+            const ALchar *resamplerName = qalGetStringiSOFT(AL_RESAMPLER_NAME_SOFT, i);
+            if (Q_stristr(resamplerName, "spline")) {
+                Com_Printf("OpenAL: Using %s as the resampler.\n", resamplerName);
+                al_resampler_index = i;
+                break;
+            }
+        }
+    }
+#endif
 
     for (i = 0; i < MAX_SOUNDSYSTEM_CHANNELS_3D; i++) {
         if (!S_OPENAL_InitChannel(i, &openal.chan_3D[i])) {

